src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -370,7 +365,16 @@ private function createFirewall(ContainerBuilder $container, $id, $firewall, &$a
                 $contextKey = $firewall['context'];
             }
 
-            $listeners[] = new Reference($this->createContextListener($container, $contextKey));
+            if (!isset($firewall['logout_on_user_change']) || !$firewall['logout_on_user_change']) {
+                @trigger_error(sprintf('Not setting "logout_on_user_change" to true on firewall "%s" is deprecated as of 3.4, it will always be true in 4.0.', $id), E_USER_DEPRECATED);
+            }
+
+            if (isset($this->firewallsByContextKey[$contextKey]) && $this->firewallsByContextKey[$contextKey][1]['logout_on_user_change'] !== $firewall['logout_on_user_change']) {
+                throw new InvalidConfigurationException(sprintf('Firewalls "%s" and "%s" need to have the same value for option "logout_on_user_change" as they are sharing the context "%s"', $this->firewallsByContextKey[$contextKey][0], $id, $contextKey));
+            }
+
+            $this->firewallsByContextKey[$contextKey] = array($id, $firewall);
+            $listeners[] = new Reference($this->createContextListener($container, $contextKey, $firewall['logout_on_user_change']));
```

Comment:
must be reached only if the key is set on `$firewall`. Also I would store the option value only instead of the whole config

Reply:
> must be reached only if the key is set on $firewall.

Not sure I follow. That line was always reached before as well?

> Also I would store the option value only instead of the whole config

:+1:

Reply:
Now it's accessing `$firewall['logout_on_user_change']` [which might be not set](https://github.com/symfony/symfony/blob/3.4/src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php#L280). 
In this case we should not register the firewall in `$firewallsByContextKey`, perhaps pass `null` to `createContextListener()` in order to not add the method call as well

Reply:
But it did that also before :wink: 

https://github.com/symfony/symfony/blob/3.4/src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php#L283

I actually think the array key will always exist and it cannot be null?

https://github.com/symfony/symfony/blob/3.4/src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php#L251

Reply:
Forgot that it's a bool node, sorry for the noise!

Reply:
No problem ;) will update the PR after work